Makefile

```diff
@@ -35,11 +39,32 @@ test:
 test-unit:
 	pipenv run ./scripts/run_tests_unit.sh
 
-test-functional:
-	pipenv run ./scripts/run_tests_functional.sh
+test-functional: generate-pages
+	yarn test_functional
+
+test-functional-headless: generate-pages
+	EQ_RUN_FUNCTIONAL_TESTS_HEADLESS='True' make test-functional
+
+test-functional-spec: generate-pages
+	yarn test_functional --spec ./tests/functional/spec/$(SPEC)
+
+test-functional-suite: generate-pages
+	yarn test_functional --suite $(SUITE)
+
+lint-js:
+	yarn lint
+
+format-js:
+	yarn format
+
+generate-spec:
+	pipenv run python -m tests.functional.generate_pages schemas/test/en/$(SCHEMA).json ./tests/functional/generated_pages/$(patsubst test_%,%,$(SCHEMA)) -r '../../base_pages' -s tests/functional/spec/$(SCHEMA).spec.js
 
 validate-test-schemas:
-	pipenv run ./scripts/validate_test_schemas.sh
+	./scripts/validate_test_schemas.sh
+
+validate-test-schema:
+	./scripts/validate_test_schemas.sh $(SCHEMA_PATH)$(SCHEMA)
```

Comment:
Super minor - but now that we have another command which adds the `.json` for you - I feel it'd be nice here too for consistency and also since they all end in `.json` I'd find it convenient to only need to type

`make validate-test-schema SCHEMA=test_address`

Reply:
Sure, I can add that in 👍

Reply:
I don't have a strong view - it was just so it would be consistent with the generate-spec command. My slight preference is consistency with how SCHEMA= is passed in but I'm still happy with it as is, and if not including the `.json` then I agree with dropping the second command